mrp_subcontracting_partner_management/models/res_partner.py

```diff
@@ -6,258 +8,219 @@ class ResPartner(models.Model):
 
     is_subcontractor_partner = fields.Boolean(string="Subcontractor")
     subcontracted_created_location_id = fields.Many2one(
-        copy=False, comodel_name="stock.location"
+        comodel_name="stock.location", copy=False
     )
     partner_picking_type_id = fields.Many2one(
-        copy=False, comodel_name="stock.picking.type"
-    )
-    partner_buy_rule_id = fields.Many2one(
-        copy=False,
-        comodel_name="stock.rule",
-    )
-    partner_resupply_rule_id = fields.Many2one(
-        copy=False,
-        comodel_name="stock.rule",
+        comodel_name="stock.picking.type", copy=False
     )
+    partner_buy_rule_id = fields.Many2one(comodel_name="stock.rule", copy=False)
+    partner_resupply_rule_id = fields.Many2one(comodel_name="stock.rule", copy=False)
+
+    CHECK_DATA = {
+        # Updating Subcontracting Location
+        "subcontracted_created_location_id": "_create_subcontracting_location_data",
+        # Updating Subcontracting operation type
+        "partner_picking_type_id": "_create_operation_type_for_subcontracting",
+        # Updating Route Rule for Subcontracting buy
+        "partner_buy_rule_id": "_create_route_rule_for_subcontracting",
+        # Updating Route Rule for Subcontracting resupply
+        "partner_resupply_rule_id": "_create_route_rule_for_subcontracting_resupply",
+    }
 
     def _set_subcontracting_values_active(self, active):
-        self.ensure_one()
-        if self.subcontracted_created_location_id:
-            self.subcontracted_created_location_id.active = active
-        if self.partner_picking_type_id:
-            self.partner_picking_type_id.active = active
-        if self.partner_buy_rule_id:
-            self.partner_buy_rule_id.active = active
-        if self.partner_resupply_rule_id:
-            self.partner_resupply_rule_id.active = active
+        """Set subcontracting values active/inactive by argument key"""
+        for key in self.CHECK_DATA:
+            self.mapped(key).write({"active": active})
 
-    def unlink(self):
+    @api.model
+    def _update_name_translation(self, records, name):
+        """Update name field translation for records"""
+        self.env["ir.translation"].search(
```

Comment:
Better to simply remove translations, isn't it?

Reply:
@pedrobaeza  When copying a vendor record or changing the vendor name then the translated name of related records doesn't change. That is why I added a translation method .

Reply:
AFAIK, standard copy method handles translations.

Reply:
@pedrobaeza If the name of a subcontractor changes, translations for the related operations types are not updated.
This function fix this behaviour for related operations types.